packages/core/ui-settings/core-ui-settings-server-internal/src/ui_settings_service.ts

```diff
@@ -78,12 +78,19 @@ export class UiSettingsService
   public async setup({ http, savedObjects }: SetupDeps): Promise<InternalUiSettingsServiceSetup> {
     this.log.debug('Setting up ui settings service');
 
+    const config = await firstValueFrom(this.config$);
+    this.overrides = config.overrides;
     savedObjects.registerType(uiSettingsType);
     savedObjects.registerType(uiSettingsGlobalType);
-    registerRoutes(http.createRouter<InternalUiSettingsRequestHandlerContext>(''));
 
-    const config = await firstValueFrom(this.config$);
-    this.overrides = config.overrides;
+    if (config.hasOwnProperty('publicApiEnabled')) {
+      registerRoutes(
+        http.createRouter<InternalUiSettingsRequestHandlerContext>(''),
+        config.publicApiEnabled
+      );
+    } else {
+      registerRoutes(http.createRouter<InternalUiSettingsRequestHandlerContext>(''));
+    }
```

Comment:
NIT: unnecessary duplication

```ts

const publicApiEnabled = config.hasOwnProperty('publicApiEnabled') && config.publicApiEnabled === true;
registerRoutes(
        http.createRouter<InternalUiSettingsRequestHandlerContext>(''),
        publicApiEnabled
);
```

Reply:
Sure, I refactored these lines of code, but I don't think the condition `config.hasOwnProperty('publicApiEnabled') && config.publicApiEnabled === true` would work for self-managed where this would evaluate to false since we don't have the `publicApiEnabled` setting configured, so I changed this logic a bit.